checkpoint.go

```diff
@@ -193,27 +193,52 @@ func Check(p *CheckParams) (*CheckResponse, error) {
 // CheckInterval is used to check for a response on a given interval duration.
 // The interval is not exact, and checks are randomized to prevent a thundering
 // herd. However, it is expected that on average one check is performed per
-// interval. The returned channel may be closed to stop background checks.
-func CheckInterval(p *CheckParams, interval time.Duration, cb func(*CheckResponse, error)) chan struct{} {
-	doneCh := make(chan struct{})
-
-	if disabled := os.Getenv("CHECKPOINT_DISABLE"); disabled != "" {
-		return doneCh
+// interval.
+// The function returns two channels:
+// 1) doneCh may be closed to stop background checks.
+// 2) infoCh may be used to send a channel to which a time at which next check
+//    will happen will be sent.
+func CheckInterval(p *CheckParams, interval time.Duration,
+	cb func(*CheckResponse, error)) (doneCh chan struct{}, infoCh chan chan time.Time) {
```

Comment:
@brb have you read https://inconshreveable.com/07-08-2014/principles-of-designing-go-apis-with-channels/?

It seems to me the use of channels here is a bit weird and not a particularly friendly API.

Have you considered how you might implement a `WhenIsNextCheck() time.Time` method instead?